src/Mod/Sketcher/Gui/CommandCreateGeo.cpp

```diff
@@ -156,6 +156,70 @@ SketcherGui::ViewProviderSketch* getSketchViewprovider(Gui::Document *doc)
+    
+    if(sug1.size()>0 && sug2.size()>0) {
+        
+        bool rmvhorvert = false;
+        
+        // we look for:
+        // 1. Coincident to external on both endpoints
+        // 2. Coincident in one endpoint to origin and pointonobject/tangent to an axis on the other
+        auto detectredundant = [psketch](std::vector<AutoConstraint> &sug, bool &ext, bool &orig, bool &axis) {
+            
+            ext = false;
+            orig = false;
+            axis = false;
+            
+            for(std::vector<AutoConstraint>::const_iterator it = sug.begin(); it!=sug.end(); ++it) {
+                if( (*it).Type == Sketcher::Coincident && ext == false) {
+                    const std::map<int, Sketcher::PointPos> coincidents = psketch->getAllCoincidentPoints((*it).GeoId, (*it).PosId);
+
+                    if(!coincidents.empty()) {
+                        ext = coincidents.begin()->first < 0; // the keys are ordered, so if the first is negative, it is coincident with external
+                        
+                        std::map<int, Sketcher::PointPos>::const_iterator geoId1iterator;
+                        
+                        geoId1iterator = coincidents.find(-1);
+                        
+                        if( geoId1iterator != coincidents.end()) {
+                            if( (*geoId1iterator).second == Sketcher::start )
+                                orig = true;
+                        }
+                    }
+                    else { // it may be that there is no constraint at all, but there is external geometry
+                        ext = (*it).GeoId < 0;
+                        orig = ((*it).GeoId == -1 && (*it).PosId == Sketcher::start);
+                    }
+                }
+                else if( (*it).Type == Sketcher::PointOnObject && axis == false) {
+                    axis = (((*it).GeoId == -1 && (*it).PosId == Sketcher::none) || ((*it).GeoId == -2 && (*it).PosId == Sketcher::none));
+                }
+                
+            }
+        };
+        
+        bool firstext = false, secondext = false, firstorig = false, secondorig = false, firstaxis = false, secondaxis = false;
+        
+        detectredundant(sug1, firstext, firstorig, firstaxis);
+        detectredundant(sug2, secondext, secondorig, secondaxis);
+        
+        
+        rmvhorvert = (   (firstext && secondext)     ||  // coincident with external on both endpoints
+        (firstorig && secondaxis)   ||  // coincident origin and point on object on other
+        (secondorig && firstaxis) );
+        
+        if(rmvhorvert) {
+            for(std::vector<AutoConstraint>::reverse_iterator it = sug2.rbegin(); it!=sug2.rend(); ++it) {
+                if( (*it).Type == Sketcher::Horizontal || (*it).Type == Sketcher::Vertical) {
+                    sug2.erase(std::next(it).base());

```

Comment:
erase() invalidates the iterator and the following call of "++" operator causes a crash in debug build on Windows.
Fixed in 5475368ea4d5